docs/DesignDocs/SealingAPI.md

```diff
@@ -0,0 +1,1171 @@
+>   * OE SDK provides `oe_seal` and `oe_unseal` APIs that are wrappers over TEE plug-in implementation.
+>
+>     * Many of the ideas from attestation plug-ins may carry over here, such as UUID targeting, plug-in registration, etc. Some will get simplified (e.g. UUID is always embedded in sealed blob because there is no cross-TEE sealing)
+>     * Some or all of the OE semantics may go away attempting to generalize sealing policies across TEEs, much in the same way that attestation plugins don't attempt to build on top of the old `oe_parse_report` semantics.
+>     * For example, consider a trivial use case where the user has no control over sealing parameters at all if they write cross-TEE code; `oe_seal` just calls the appropriate TEE-plugin and it seals with a default policy of its choice (e.g. always with MRENCLAVE for SGX). The TEE-plugin may then expose its custom set of properties through an opaque blob type through the `oe_seal` API that a user that wants to customize functionality on a specific TEE can manipulate themselves (e.g. `sgx_key_request` struct for SGX).
+>
+> I'm not proposing that we actually run off and go build this strawman (there are some other problems I'm glossing over here like the `oe_get_private_key` API) but it represents a high level concept for an alternative that I think hits some of the stated objectives better:
+>
+> * Actually TEE-agnostic, bakes less implementation directly into OE SDK (or at least, could be more easily broken out as standalone components moving forwards)
+> * Accommodative in that the plug-ins get to expose the full range of options specific to a TEE if desired without it filtering through an OE SDK aggregation layer.
+>
+>   * The trade-off being less expressive sealing semantics for apps that want to span different TEEs, at least in this strawman version.
+> * Easy to use in that there's no management of key info at all, and that's incorporated into the `oe_seal` function.
+>
+>   * With proper built-in defaults per TEE, it could also avoid some of the registration complexity associated with attestation plugins, again because there's a much stronger binding of TEE the enclave is running in vs. attestation.
+> * Interoperable with existing Intel SDK, since Intel controls the plugin entirely, they can choose to provide that functionality as they see fit.
+
+Plug-ins allows multiple implementations to coexist and allows the developer to choose which plug-ins to go into the enclave. Based on the analyses in previous sections, plug-ins are not necessary IMHO.
+
+- [ ] (**Decision?**) Plug-in model
+
+### How to specify seal keys?
+
+There are 3 options to specify seal key for `oe_seal()`.
+
+- [ ] `oe_seal()` takes all necessary parameters for deriving `key_info` and the seal key.
+  - This gives users a one-stop experience.
+  - But it won't work for (TEE specific) features not exposed as a parameter.
```

Comment:
This strikes me as creating an incorrect conclusion, since there are clearly more options available for parameterizing `oe_seal()` than indicated by the specific strawman presented. It's not a convincing argument that exposing the `key_info` modified from the existing internal OE data structure is the only way to address this issue. See alternative `oe_seal()` proposal above.